src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -39,15 +55,26 @@ def consume(self, state):
             logger.debug('Moving to previous replication log offset: %s (current position: %s)...', offset, current)
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
+        else:
+            decoded = prime_for_batch_start(
```

Comment:
Probably worth adding a comment here explaining this, since it's probably a little weird at first glance.